src/pages/change-password/index.tsx

```diff
@@ -1,65 +1,84 @@
 import React, { useState } from "react"
 import SignInTextField from "@/components/text-fields/sign-in-text-field"
 import MainButton from "@/components/buttons/main-button"
-import ErrorPopup from "@/components/popups/error-popup"
 import { useSupabaseClient, useUser } from "@supabase/auth-helpers-react"
 import { Database } from "@/types/database"
 import { setNameAndFirstLoginFalse } from "@/services/authentication.service"
 import { useRouter } from "next/router"
+import { RegisterOptions, useForm } from "react-hook-form"
+import ErrorMsg from "@/components/error/error-msg"
+import {ChangePasswordInputs, SignInInputs} from "@/types/form-types";
 
+export interface FormDataItem {
+  tag: ChangePasswordInputs | SignInInputs
+  type: string
+  placeholder: string
+  dataCy: string
+  validationOptions: RegisterOptions
+}
 const Index = () => {
-  // TODO Use react-hook-form for this for cleaner code
   const supabase = useSupabaseClient<Database>()
   const user = useUser()
-  const [password, setPassword] = useState<string>("")
-  const [name, setName] = useState<string>("")
-  const [confirmPassword, setConfirmPassword] = useState<string>()
-  const [errorText, setErrorText] = useState<string>("")
-  const [showErrorText, setShowErrorText] = useState<boolean>(false)
   const router = useRouter()
+  const [errorMsg, setErrorMsg] = useState("")
+  const [showError, setShowError] = useState(false)
+  const {
+    handleSubmit,
+    register,
+    watch,
+    formState: { errors },
+  } = useForm()
+  const password = watch("password")
+  const name = watch("userName")
 
-  const setError = (errorText: string) => {
-    setErrorText(errorText)
-    setShowErrorText(true)
-  }
-
+  const changePasswordFormData: FormDataItem[] = [
+    {
+      tag: "userName",
+      type: "text",
+      placeholder: "Username",
+      dataCy: "set-name-textfield",
+      validationOptions: { required: "Username is required" },
```

Comment:
Change username to name since it is not necessarily a username but just their IRL name